Azure node replacement document modification (#198)
scalar-labs/scalar-terraform: docs/CassandraOperation.md

When you taint the volume attachment terraform will try to attach the same data or commit log volume to the new instance. This is the ideal situation as it is the quickest way to replace a node.

##### Azure

> If crashed node is available in resource group

Reviewer: What do you mean `a crashed node` is `available`?
Can you clarify and update the sentence correctly?

Author: Thank you for the review comments.
> What do you mean a crashed node is available?

I mean, Nodes are listed in the Azure resource group after crashing

> Can you clarify and update the sentence correctly?

Sentence updated like `If the crashed node is listed in the resource group`

When you taint the volume attachment terraform will try to attach the same data or commit log volume to the new instance. This is the ideal situation as it is the quickest way to replace a node.

##### Azure

> If the crashed node is listed in the resource group

Reviewer: ```suggestion
```

```console
terraform taint "module.cassandra.module.cassandra_cluster.azurerm_virtual_machine.vm-linux[0]"
terraform taint "module.cassandra.azurerm_virtual_machine_data_disk_attachment.cassandra_data_volume_attachment[0]"

terraform apply
```

> If you face an error or accidentally terminated the node, please try the following.

Reviewer: ```suggestion
If you face an error or accidentally terminated the node or deleted the OS disk manually, please try the following.
```

Author: Sorry, I am not sure about this because I couldn't delete os disk manually without terminating VM.

Reviewer: OK. The point is that the documentation is too specific to your test case; terminate an instance.
It should be generalized since users don't really terminate it.

> If the crashed node is not listed in the resource group

Reviewer: ```suggestion
If you face an error or accidentally terminated the node, please try the following.
* Remove an OS disk if attached to the node.
* Do `terraform state rm` as follows.
```

> 

Reviewer 2: I'm not sure if this is possible in the cloud, Maybe It's most likely due to operation mistake.

Author: Sorry, I'm not sure if this is necessary or not.
We can use these steps when the Cassandra node is accidentally terminated.

We can't replace the Cassandra node with the `terraform taint` command when the c* node is terminated.

Reviewer 2: >We can use these steps when the Cassandra node is accidentally terminated

It doesn't seem to related with crashing.

Can we just consider these two cases?

- Basically, Can replace nodes with `taint`.
- When the Cassandra node is accidentally terminated, Need to remove os-disk and remove state.

Reviewer: @scalar-boney As Tei-san says, it might be weird to explain like that.

Can you clarify what would happen if you don't terminate the instance and taint it?
(the added steps are not needed?)

Author: Sorry for the inconvenience. 
It is modified more conveniently with the help of Yamada san.
Could you please check once again?

> * Remove an OS disk if attached to the node.

Reviewer: ```suggestion
* Terminate the node and remove the OS disk if it is attached to the node.
```

Author: It is a bit confusing for me,

Can we change like this
* Terminate the node if the OS disk is deleted manually
* Remove the OS disk if it is attached to the terminated node

> * Remove os-disk of crashed node If it is available.

Reviewer: ```suggestion
```
